Building with cmake's fetch content using a cross compiler

So I have the following in my CMakeLists.txt file:
```
include(FetchContent)

set(FMT_TEST FF CACHE INTERNAL "disabling fmt tests")

FetchContent_Declare(
        fmt
        GIT_REPOSITORY  https://github.com/fmtlib/fmt.git
        GIT_TAG         8.0.1
        GIT_PROGRESS    TRUE
        USES_TERMINAL_DOWNLOAD TRUE
)

FetchContent_MakeAvailable(fmt)

...
target_link_libraries(mybinary
        PRIVATE fmt::fmt
)

```

When cross-compiling with say an arm compiler, i get undefined reference errors like this:
```
<artificial>:(.text.startup+0x1588): undefined reference to `fmt::v8::vprint(fmt::v8::basic_string_view<char>, ...
```

I'm cross-compiling like this:
```
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=arm-openwrt-linux-gcc -DCMAKE_CXX_COMPILER=arm-openwrt-linux-g++ ..
```
So i'm not specifying a toolchain file.

Now, is this a cmake bug in the FetchContent module, or is this a libfmt bug?

Cheers

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -251,7 +251,7 @@ endif ()
 if (MINGW)
   check_cxx_compiler_flag("-Wa,-mbig-obj" FMT_HAS_MBIG_OBJ)
   if (${FMT_HAS_MBIG_OBJ})
-    target_compile_options(fmt PUBLIC "-Wa,-mbig-obj")
+    target_compile_options(fmt PRIVATE "-Wa,-mbig-obj")
   endif()
 endif ()
 

